packages/iov-tendermint-rpc/src/adaptor.ts

```diff
@@ -32,6 +32,10 @@ export interface Params {
   readonly encodeGenesis: (req: requests.GenesisRequest) => JsonRpcRequest;
   readonly encodeHealth: (req: requests.HealthRequest) => JsonRpcRequest;
   readonly encodeStatus: (req: requests.StatusRequest) => JsonRpcRequest;
+  readonly encodeSubscribe: (
```

Comment:
Hmmm... you have the following types:

```
export interface SubscribeRequest {
  readonly method: Method.SUBSCRIBE;
  readonly type: SubscriptionEventType;
}

export interface SubscribeRequestQuery {
  readonly tags: ReadonlyArray<Tag>;
}
```

This is different than all the others, that keep a format closer to the wire. I understand that there is some string building going on here, so it is not as clear cut, but why not something like:

```
export interface SubscribeRequest {
  readonly method: Method.SUBSCRIBE;
  readonly query: {
    readonly type: SubscriptionEventType;
    readonly tags?: ReadonlyArray<Tag>;
  }
}
...
readonly encodeSubscribe: (req: requests.SubscribeRequest) => JsonRpcRequest;
```

One is not "better", but the second seems more consistent with the rest